lib/src/main/kotlin/com/lemonappdev/konsist/api/ext/sequence/KoClassDeclarationSequenceExt.kt

```diff
@@ -210,15 +215,35 @@ fun Sequence<KoClassDeclaration>.withSomeParents(name: String, vararg names: Str
 }
 
 /**
- * Sequence containing all classes that don't have parents.
+ * Sequence containing all classes that have no parent.
+ *
+ * @return A sequence containing classes that have no parent.
+ */
+fun Sequence<KoClassDeclaration>.withoutParents(): Sequence<KoClassDeclaration> = filterNot { it.hasParents() }
+
+/**
+ * Sequence containing all classes that don't have all specified parents.
  *
+ * @param name The name of the parent to exclude.
  * @param names The name(s) of the parent(s) to exclude.
- * @return A sequence containing classes that don't have the specified parent(s) (or none parent if [names] is empty).
+ * @return A sequence containing classes that don't have all specified parent(s).
  */
-fun Sequence<KoClassDeclaration>.withoutParents(vararg names: String): Sequence<KoClassDeclaration> = filter {
-    when {
-        names.isEmpty() -> !it.hasParents()
-        else -> !it.hasParents(*names)
+fun Sequence<KoClassDeclaration>.withoutAllParents(name: String, vararg names: String): Sequence<KoClassDeclaration> = filter {
+    !it.hasParents(name, *names)
+}
+
+/**
+ * Sequence containing all classes that don't have some parents.
+ *
+ * @param name The name of the parent to exclude.
+ * @param names The names of the parents to exclude.
+ * @return A sequence containing classes that don't have at least one of the specified parent(s).
+ */
+fun Sequence<KoClassDeclaration>.withoutSomeParents(name: String, vararg names: String): Sequence<KoClassDeclaration> = filter {
+    !it.hasParents(name) && if (names.isNotEmpty()) {
+        names.any { name -> !it.hasParents(name) }
+    } else {
+        true
     }
 }
 
```

Comment:
IN this method and all methods that have `name` or `names` arguments change "parent of type" to "names parents".

Same for "parent interface of type" -> "named parent interface"